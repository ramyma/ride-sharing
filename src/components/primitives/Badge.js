import styled from '@xstyled/emotion';
import Typography from './Typograpgy';

const Badge = styled(Typography)``;

Badge.defaultProps = {
  backgroundColor: 'primary',
  borderRadius: 4,
  color: 'white',
  fontSize: 1,
  textAlign: 'center',
  p: 1,
  minWidth: 9
};

export default Badge;
