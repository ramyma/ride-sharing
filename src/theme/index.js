import { opacify } from 'polished';

const theme = {
  breakpoints: {
    xs: 0,
    sm: 576,
    md: 768,
    lg: 992,
    xl: 1200
  },
  defaultColorModeName: 'light',
  colors: {
    background: '#fff',
    grey: '#585858',
    inverseBackground: '#1c1c1c',
    shadow: opacify(0.19, '#000'),
    textDark: '#353535',
    text: '#353535',
    primary: '#ee424a',
    success: '#4CAF50',
    danger: '#f44336',
    warning: '#ffa000',
    modes: {
      dark: {
        background: '#1c1c1c',
        grey: '#9e9e9e',
        inverseBackground: '#fff',
        shadow: opacify(0.19, '#fff'),
        text: '#fff'
      }
    }
  },
  fontSizes: [0, 12, 14, 16, 20, 24, 32, 48, 64, 72],
  sizes: [0, 2, 4, 8, 10, 12, 14, 16, 18, 20, 22, 24, 26, 28, 30, 32],
  shadows: {
    grey: 'rgba(0, 0, 0, 0.19) 0 0 10px 0px',
    greyInset: `inset rgba(0, 0, 0, 0.19) 0 0 10px 0px`,
    pink: '#e42974 0 0 3px',
    text: '0 0 rgba(0, 0, 0, 0.19)'
  },
  space: [0, 4, 8, 16, 24, 48, 96, 144, 192, 240]
};

export default theme;
