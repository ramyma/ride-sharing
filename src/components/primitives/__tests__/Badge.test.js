import React from 'react';
import renderer from 'react-test-renderer';
import { createSerializer } from 'jest-emotion';
import * as emotion from '@xstyled/emotion';
import { ThemeProvider } from '@xstyled/emotion';
import Badge from '../Badge';
import theme from '../../../theme';

expect.addSnapshotSerializer(createSerializer(emotion));

it('Badge renders correctly', () => {
  const tree = renderer
    .create(
      <ThemeProvider theme={theme}>
        <Badge>1</Badge>
      </ThemeProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
