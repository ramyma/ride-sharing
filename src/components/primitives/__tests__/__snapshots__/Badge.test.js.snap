// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge renders correctly 1`] = `
.emotion-0 {
  background-color: #ee424a;
  border-radius: 4px;
  color: white;
  min-width: 20px;
  padding: 4px;
  color: #353535;
  font-size: 14px;
  color: white;
  font-size: 12px;
  text-align: center;
}

<div
  className="emotion-0"
  color="white"
  fontSize={1}
>
  1
</div>
`;
